### DISCUSSION TITLE
Lesson 6: Not getting "Insufficient Funds" error

### DISCUSSION THREAD

**Discussion by ekong16** [smartcontractkit/full-blockchain-solidity-course-py#280] [Q&A] [answered]
Hello,
I am going through the tutorial and I am on lesson 6. When I try to run the deploy script on the mainnet-fork, I am not getting the "Insufficient Funds" error like in the video. It seems to work fine even though my mainnet balance is 0. Just wondering if anyone has any thoughts on why this is...

Here is the output:

 ```
% brownie run scripts/deploy.py --network mainnet-fork
Brownie v1.16.4 - Python development framework for Ethereum

BrownieFundMeProject is the active project.

Launching 'ganache-cli --accounts 10 --hardfork istanbul --fork https://mainnet.infura.io/v3/<redacted> --gasLimit 12000000 --mnemonic brownie --port 8545 --chainId 1'...

Running 'scripts/deploy.py::main'...
balance:  0
Price feed address:  0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419
Active network:  mainnet-fork
Transaction sent: 0x79af9c2555dab732d7bc1f9774b7598aca174ca5a2dcfe02cd31ae19af8ae6ae
  Gas price: 0.0 gwei   Gas limit: 12000000   Nonce: 0
  FundMe.constructor confirmed   Block: 13512364   Gas used: 405345 (3.38%)
  FundMe deployed at: 0x2Da886635524809487d0c2cC673D9CccB7E24a64

Contract deployed to 0x2Da886635524809487d0c2cC673D9CccB7E24a64
Terminating local RPC client...

```

**Comment by cromewar** [comment, marked as answer]
Hello @ekong16 this could happen because the `get_account` function... you might have something like this:

```Python
def get_account():
    if (
        network.show_active() in LOCAL_BLOCKCHAIN_ENVIRONMENTS
        or network.show_active() in FORKED_LOCAL_ENVIRONMENTS
    ):
        return accounts[0]
    else:
        return accounts.add(config["wallets"]["from_key"])
```
so as mainnet-fork is on FORKED_LOCAL_ENVIRONMENTS this will return the first account from ganache-cli as you can see on your run ourtput.

![image](https://user-images.githubusercontent.com/20199180/139447941-9b8b413b-957e-4566-b816-ea48071a3449.png)

I might be taking the gas needed for the transaction from ganache.

Hope this info might help.

**Reply by ekong16** [reply]
Thank you for the response. 

I think it is using my actual Ethereum address and not a local ganache account. 

Here is my get_account function:

```
def get_account():
    if network.show_active() in LOCAL_BLOCKCHAIN_ENVIRONMENTS:
        return accounts[0]
    else:
        return accounts.add(config["wallets"]["from_key"])`
```

**Reply by cromewar** [reply]
Wow that's curious according to Rinkeby Etherscan here is your contract:

https://rinkeby.etherscan.io/address/0x2Da886635524809487d0c2cC673D9CccB7E24a64

and as you can see you are paying gas for the transaction, are you 100% sure you don't have any ETH on your rinkeby wallet?

BTW - Did you deployed the contract 19 days ago?

**Reply by ekong16** [reply]
Oh wow interesting I didn't know I was deploying to rinkeby. I definitely have ETH in the Rinkeby wallet. I just don't have ETH in mainnet. I thought it was supposed to deploy to mainnet fork and not rinkeby? 

I think the etherscan link you are showing are contracts I deployed a while ago from earlier lessons....

**Reply by cromewar** [reply]
Actually I didn't know either, but seems the most logical reason... because mainnet and mainnet-fork are not the same, if you type `brownie networks list` you should see that mainnet-fork is on the the development networks, which should use ganache...

however I used the contract you provided here: 

![image](https://user-images.githubusercontent.com/20199180/139456940-dc80ce35-4145-4315-8013-50f0b768eaf9.png)

And it seems to be deployed on Rinkeby.

**Reply by cromewar** [reply]
However take something in consideration, if you deploy a contract to different networks as the bytecode is the same,  you would have this hash code 0x2Da886635524809487d0c2cC673D9CccB7E24a64 on any network, so I suppose is more probable  the contract have been deployed to rinkeby earlier (by Patrick) and that's why it says it was deployed 19 days ago.

So for mainnet-network I think brownie is omitting this return accounts.add(config["wallets"]["from_key"])` and using ganache for the gas fees.

Based on your terminal I'm going to suppose you are using Linux, so, could you please export again your env variables? remember:

export PRIVATE_KEY=blablabla

and the same with the others.

Finally make sure your configuration file is `brownie-config.yaml` and not `brownie_config.yaml` or anything else. 

Give it a try.

**Reply by ekong16** [reply]
I think what's interesting is that the gas price is 0.0 gwei. 

All the code is set up correctly (env variables, accounts, etc.) 

But for some reason deploying to the mainnet-fork is not charging any gas fees for me.

**Reply by cromewar** [reply]
did you check the config file name? 
This is very interesting and worth to be documented, please go ahead with lesson 7 and try again with mainnet-fork, and let's see what happens

**Comment by washbin** [comment]
Hello, seems like this discussion hasn't yet been resolved. 
I reached the same part today and faced the exact same issue.

**Reply by cromewar** [reply]
Give a a try changing something inside the contract, like a variable name or so. This might change the hashcode. Also remember that 0.0 gwei does no mean 0 wei, as it has 18 decimals.

**Reply by MananJain2002** [reply]
Hi the only reason this is happening is because our mainnet-fork is running on Ganache-CLI unlike for patrick's which is running on different environment
![Ganache-CLI](https://user-images.githubusercontent.com/63538502/165140032-973b79ff-0f21-4da9-809b-27f8c361cc39.png)
![mainnet-fork](https://user-images.githubusercontent.com/63538502/165140054-10ab790d-5a91-4bce-856a-c1475be0e950.png)

**Comment by alexkalen** [comment]
So? Any suggetions on what to do? Should we just skip till Lesson 7?

**Comment by Python-David** [comment]
Lending my voice to this: waiting for a resolution. Thanks @MananJain2002 for astutely pointing that out

**Comment by Niferu** [comment]
Hello I'm having same "issue"(@MananJain2002 is totally right) and I believe we can just skip this part and go straight for setuping our own for example Alchemy mainnet and it will be all good. Correct me if I'm wrong
